From c9f8ba2091da223fa703923e36afa4ba5f1731aa Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Wed, 23 Jun 2021 19:06:12 -0400
Subject: [PATCH] Maintain current contextIsolation behavior across electron 11
 vs. >=12

For what this does, see:

https://www.electronjs.org/docs/breaking-changes#default-changed-contextisolation-defaults-to-true
https://github.com/electron/electron/blob/main/docs/tutorial/context-isolation.md

In electron 11, this logged:

> The default of contextIsolation is deprecated and will be changing from
> false to true in a future release of Electron.

In electron 12, Keybase refused to start. Explicitly specifying the old
behavior makes Keybase run again.

"for the security of your application" Keybase may wish in future to
restructure and support context isolation, but that would be in the
future.
---
 shared/desktop/app/main-window.desktop.tsx | 1 +
 shared/desktop/app/menu-bar.desktop.tsx    | 1 +
 shared/desktop/app/node.desktop.tsx        | 1 +
 3 files changed, 3 insertions(+)

diff --git a/shared/desktop/app/main-window.desktop.tsx b/shared/desktop/app/main-window.desktop.tsx
index a93ffa11a7..9ef4f6f288 100644
--- a/shared/desktop/app/main-window.desktop.tsx
+++ b/shared/desktop/app/main-window.desktop.tsx
@@ -275,6 +275,7 @@ export default () => {
     minWidth: 740,
     show: false,
     webPreferences: {
+      contextIsolation: false,
       backgroundThrottling: false,
       devTools: showDevTools,
       enableRemoteModule: true,
diff --git a/shared/desktop/app/menu-bar.desktop.tsx b/shared/desktop/app/menu-bar.desktop.tsx
index 1597c020bd..e1aa62a854 100644
--- a/shared/desktop/app/menu-bar.desktop.tsx
+++ b/shared/desktop/app/menu-bar.desktop.tsx
@@ -40,6 +40,7 @@ export default (menubarWindowIDCallback: (id: number) => void) => {
       resizable: false,
       transparent: true,
       webPreferences: {
+        contextIsolation: false,
         enableRemoteModule: true,
         nodeIntegration: true,
         nodeIntegrationInWorker: false,
diff --git a/shared/desktop/app/node.desktop.tsx b/shared/desktop/app/node.desktop.tsx
index 6368bc18b9..7c8c4a34d5 100644
--- a/shared/desktop/app/node.desktop.tsx
+++ b/shared/desktop/app/node.desktop.tsx
@@ -365,6 +365,7 @@ const plumbEvents = () => {
           show: false, // Start hidden and show when we actually get props
           titleBarStyle: 'customButtonsOnHover' as const,
           webPreferences: {
+            contextIsolation: false,
             enableRemoteModule: true,
             nodeIntegration: true,
             nodeIntegrationInWorker: false,
-- 
2.32.0

